### ISSUE TITLE
Built-in SSL / TLS termination 

### ISSUE THREAD

**Issue by timlohse1104** [mudler/LocalAI#1295]
**Is your feature request related to a problem? Please describe.**
When thinking about using localAI in a production environment to serve an open source llm with an OpenAI compatible API things like SSL / TLS termination come to mind. I searched [the webpage](https://localai.io/) and [github repository](https://github.com/mudler/LocalAI/) for built-in support in localAI but could not find any appearences. Maybe i missed something but i would like to ask before implementing a reverse proxy capable on this task. 

**Describe the solution you'd like**
An launch option for built-in SSL / TLS termination in localAI

**Describe alternatives you've considered**
Architectural approach using an reverse proxy like nginx

**Additional context**
We would like to use localAI in a production context connecting to a llama.cpp backend service and serving a "like OpenAI" API for an open source llm.

**Comment by hotspoons** [comment]
@timlohse1104 if you deploy this using the helm chart, it is as simple as adding a cluster issuer annotation to the values.yaml file and defining your ingress, provided you configured a cluster CA:

![image](https://github.com/mudler/LocalAI/assets/851973/42bd98a7-e89c-46dc-8050-c6973393caf5)
![image](https://github.com/mudler/LocalAI/assets/851973/e64ebf64-d1d1-4e74-a09c-23d4d04dc915)

If a kubernetes cluster doesn't make sense for your use case, using a reverse proxy is an easy alternative - Envoy is super easy to get working for SSL termination, nginx is a little more involved.

**Comment by pritchey** [comment]
Having built-in support for HTTPS within the LocalAI server is something I'm interested in as well.  While using a reverse proxy technically does provide HTTPS support, the communication on loopback is still unencrypted and in some environments will not be acceptable during a security audit or by policy.  Many organizations require encryption regardless if the traffic is transmitted over a publicly accessible IP address/port, or over the internal (not publicly accessible) loopback interface.

**Comment by lee-b** [comment]
Agree.  k8s ingress is nice and all, but (a) not everyone will be using k8s, and (b) to do it right in k8s you really need a either what this ticket asks for, or a service-mesh-like sidecar; ingress is still insufficient for zero trust.
